core/controllers/access_validators.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the validating access."""
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from constants import constants
+from core.controllers import acl_decorators
+from core.controllers import base
+from core.domain import classroom_services
+from core.domain import user_services
+
+import feconf
+
+from typing import Any # pylint: disable=unused-import
+from typing import Dict # pylint: disable=unused-import
+from typing import Text # pylint: disable=unused-import
+
+
+# TODO(#13605): Refactor access validation handlers to follow a single handler
+# pattern.
+
+class SplashAccessValidationHandler(base.BaseHandler):
+    """When a request is made to '/', check the user's login status, and
+    redirect them appropriately.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    URL_PATH_ARGS_SCHEMAS = {} # type: Dict[Text, Any]
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {}
+    } # type: Dict[Text, Any]
+
+    @acl_decorators.open_access # type: ignore[misc]
+    def get(self):
+        # type: () -> None
+        if self.user_id and user_services.has_fully_registered_account( # type: ignore[no-untyped-call]
+                self.user_id):
+            user_settings = user_services.get_user_settings(self.user_id) # type: ignore[no-untyped-call]
+            default_dashboard = user_settings.default_dashboard
+            self.render_json( # type: ignore[no-untyped-call]
+                {'valid': False, 'default_dashboard': default_dashboard})
+        else:
+            self.render_json({'valid': True}) # type: ignore[no-untyped-call]
```

Comment:
Can you add a comment what 'valid' means here? Because just looking at the code, I'd assume both are 'valid' because technically, there is no issue with both requests.

Reply:
The user intended to visit the splash page by visiting route `/`. But he is logged in the first case, so he needs to be redirected to the preferred dashboard (in other words his request to visit / route is not valid).
I have added a comment.